tests/lib/prepare.sh

```diff
@@ -532,6 +532,43 @@ EOF
     rm -rf "$UNPACK_DIR"
 }
 
+# Builds kernel snap with bad kernel.efi, in different ways
+# $1: snap we will modify
+# $2: target folder for the new snap
+# $3: argument, type of corruption we want for kernel.efi
+uc20_build_corrupt_kernel_snap() {
+    local ORIG_SNAP="$1"
+    local TARGET_DIR="$2"
+    local optArg=${3:-}
+
+    # kernel snap is huge, unpacking to current dir
+    local REPACKED_DIR=repacked-kernel
+    local KERNEL_EFI_PATH=$REPACKED_DIR/kernel.efi
+    unsquashfs -d "$REPACKED_DIR" "$ORIG_SNAP"
+
+    case "$optArg" in
+        --empty)
+            printf "" > "$KERNEL_EFI_PATH"
+            ;;
+        --zeros)
```

Comment:
nitpick: zeroes

Reply:
Both are correct apparently: https://oneminuteenglish.org/en/zeros-or-zeroes/ 🙂

Reply:
Nice, one never stops learning! But my favourite quote from that page is "Zeros is more popular than Zeros but you can choose either." :-)